### PR TITLE
Implement ToCss for selector types

### PR DESCRIPTION
**Pull request by SimonSapin**
r? @emilio

---

This change is [Reviewable](https://reviewable.io/reviews/servo/rust-selectors/96)

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -985,7 +1255,7 @@ pub mod tests {
             pseudo_element: Some(PseudoElement::Before),
             specificity: specificity(0, 0, 1),
         })));
-        assert_eq!(parse("div :after"), Ok(vec!(Selector {
+        assert_eq!(parse("div ::after"), Ok(vec!(Selector {
```
Comment (emilio) [inline src/parser.rs:1258]: Is this change meaningful, or just a refactoring/nit?

**Comment by emilio** [comment]
The overall approach looks good to me, I'd like to see more testing of this, though I guess if we're going to expose that via CSSOM, we'll have more test-coverage there.

I think that a few unit tests in this repo would be great still, though. I need to review a bit more carefully, but LGTM.

**Comment by SimonSapin** [comment]
The line:

``` rust
assert_eq!(selectors[0].to_css_string(), input);
```

applies to existing unit tests and makes sure that serializing the parsed selector gives the same string as the original parsing input. So there is _some_ testing of the new code.

`:after` (single colon) is parsed as a pseudo-element per the CSS2 special case and so is the same as `::after` (double colon), but the serializer always uses the new double colon syntax so I changed the input to match. Another option would be to specify the expected serialization as a separate string.

**Comment by emilio** [comment]
Ohh missed that, great!

**Comment by larsbergstrom** [comment]
@bors-servo retry

I know this passed (https://travis-ci.org/servo/rust-selectors/builds/151305654), but I just rolled out a potential homu fix and want to see if the autolander is autolanding again :-)

**Pull request merged**